conans/client/plugin_manager.py

```diff
@@ -0,0 +1,89 @@
+import traceback
+import os
+import sys
+import uuid
+
+from conans.client.output import ScopedOutput
+from conans.errors import ConanException, NotFoundException
+from conans.tools import chdir
+
+
+class PluginManager(object):
+
+    def __init__(self, plugins_folder, plugin_names, output):
+        self._plugins_folder = plugins_folder
+        self._plugin_names = plugin_names
+        self.plugins = {}
+        self.output = output
+
+    def execute(self, method_name, **kwargs):
+        if not self.plugins:
+            self.load_plugins()
+
+        for name, plugin in self.plugins.items():
```

Comment:
This is about performance (so you feel free to ignore it as we are not going to have so many plugins, are we?): the `self.plugins` dictionary could be organized in a different way, 

```python
self.plugins = defaultdict(list)

def _register_plugin(self, plugin_obj):
    for method in dir(plugin_obj):  # or whatever gives you all the functions
        if method in list_of_valid_methods_names:
            self.plugins[method.__name__].append((plugin_obj, method))
```

and then

```python
def execute(self, method_name, **kwargs):
    assert method_name in list_of_valid_methods_names, "Who called this and from where?!"
    for plugin, method in self.plugins[method_name]:
        ...
        method(output, **kwargs)
```

I also like the idea of having in a centralized place the set of functions that a plugin can define (`list_of_valid_method_names`).

Reply:
Yes. this would be great, should be done. Can be done/optimized in a second iteration, not necessarily now.